### PR TITLE
Promote ReplicationController e2e test to Conformance

### LINKED ISSUE #78834: Write Scale Tests for Conformance
Using the list of tested stable endpoints referenced in https://github.com/kubernetes/kubernetes/issues/78748, we can pick out the endpoints which are part of the Scale kind:

[replaceCoreV1NamespacedReplicationControllerScale](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-replaceCoreV1NamespacedReplicationControllerScale&showUntested=false&showConformanceTested=false)

[readCoreV1NamespacedReplicationControllerScale](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-readCoreV1NamespacedReplicationControllerScale&showUntested=false&showConformanceTested=false)

These are stable core endpoints which are covered by an existing test, but the test does not directly exercise these endpoints. New tests should be written.

/area test
/area conformance
/sig testing

### PR DESCRIPTION
**Pull request by devaips**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Promotes the following E2E test to Conformance:
 `test/e2e/common/lifecycle.go: "should be able to delete nodes"`

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/78834

**Special notes for your reviewer**:

Part of Umbrella Issue https://github.com/kubernetes/kubernetes/issues/78748

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/area conformance
/area test

@kubernetes/sig-architecture-pr-reviews
@kubernetes/sig-cluster-lifecycle-pr-reviews 
@kubernetes/cncf-conformance-wg

### CODE REVIEW

File: test/e2e/lifecycle/resize_nodes.go [hunk 1]
```diff
@@ -107,7 +107,12 @@ var _ = SIGDescribe("Nodes [Disruptive]", func() {
 			framework.ExpectNoError(err)
 		})
 
-		ginkgo.It("should be able to delete nodes", func() {
+		/*
+		Release : v1.16
+		Testname: ReplicationController: Delete Nodes
+		Description: Ensure that pods are recreated by the replication controller after nodes are deleted.
+	*/
+		framework.ConformanceIt("should be able to delete nodes", func() {
```
Comment (draveness) [inline test/e2e/lifecycle/resize_nodes.go]: /ok-to-test
/priority important-soon

**Comment by smarterclayton** [comment]
/hold

This is a disruptive test - promoting it to conformance would mean that all conformant implementations will have to support deletion of nodes which may not be within the bounds of what a particular service allows.

This is somewhat novel, and I think needs to be discussed further before we set the precedent (either in the WG or here).

**Comment by spiffxp** [comment]
nit: if it's disruptive it should be tagged as such, we wouldn't be allowing the test in as conformance

**Comment by spiffxp** [comment]
/close
The [Disruptive] tag was present, in the top-level SIGDescribe.  The SIGDescribe > Describe > It hierarchy isn't getting picked up by the conformance walker when it tries to evaluate the full test name

File: test/conformance/testdata/conformance.txt [hunk 2]
```diff
@@ -192,6 +192,7 @@ test/e2e/kubectl/kubectl.go: "should update a single-container pod's image"
 test/e2e/kubectl/kubectl.go: "should create a job from an image, then delete the job"
 test/e2e/kubectl/kubectl.go: "should support proxy with --port 0"
 test/e2e/kubectl/kubectl.go: "should support --unix-socket=/path"
+test/e2e/lifecycle/resize_nodes.go: "should be able to delete nodes"
```
Comment (spiffxp) [inline test/conformance/testdata/conformance.txt]: This should have not been allowed in the lest and should have failed our conformance list verification test

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/kubernetes/issues/79344

**Comment by spiffxp** [comment]
https://github.com/kubernetes/kubernetes/pull/79405 should prevent these sorts of tests from getting promoted in the future

**Pull request closed without merging**